2024-03-16T15:01:00.5179990Z /Users/runner/work/_temp/mariadb-10.11.7/extra/mariabackup/xtrabackup.cc:2431:39: error: use of undeclared identifier 'OS_DATA_FILE_NO_O_DIRECT'
2024-03-16T15:01:00.5282430Z                                       OS_DATA_FILE_NO_O_DIRECT, false, &ret);
2024-03-16T15:01:00.5385590Z                                       ^
2024-03-16T15:01:01.4599970Z 1 error generated.

diff --git a/extra/mariabackup/xtrabackup.cc b/extra/mariabackup/xtrabackup.cc
index 1482c5f3507..8379a51ea5a 100644
--- a/extra/mariabackup/xtrabackup.cc
+++ b/extra/mariabackup/xtrabackup.cc
@@ -2418,7 +2418,12 @@ static bool innodb_init()
   os_file_delete_if_exists_func(ib_logfile0.c_str(), nullptr);
   os_file_t file= os_file_create_func(ib_logfile0.c_str(),
                                       OS_FILE_CREATE, OS_FILE_NORMAL,
-                                      OS_DATA_FILE_NO_O_DIRECT, false, &ret);
+#if defined _WIN32 || defined HAVE_FCNTL_DIRECT
+                                      OS_DATA_FILE_NO_O_DIRECT,
+#else
+                                      OS_DATA_FILE,
+#endif
+                                      false, &ret);
   if (!ret)
   {
   invalid_log:
